### ISSUE TITLE
Lab shuffle: replace Array with mutable.ArraySeq

### ISSUE THREAD

**Issue by bjornregnell** [lunduniversity/introprog#492]
ArraySeq behaves like collections should, which Array doesn't... 
And they are almost equally fast if not used fro primitives, sometimes ArraySeq is actually faster that Array...
https://stackoverflow.com/questions/5028551/array-vs-arrayseq-comparison.

It would be great if someone would like to make performance comparison for the shuffle lab and see how mutable.ArraySeq compares to Array...

**Comment by bjornregnell** [comment]
I guess we should not update this year, but still good if someone can check performance so I know what to say on next weeks lectures about that...

**Comment by bjornregnell** [comment]
As background info: More on performance here (but not including ArraySeq): https://www.lihaoyi.com/post/BenchmarkingScalaCollections.html

**Comment by bjornregnell** [comment]
https://docs.scala-lang.org/overviews/core/collections-migration-213.html  ctrl+f arrayseq

**Comment by bjornregnell** [comment]
https://docs.scala-lang.org/overviews/collections-2.13/performance-characteristics.html

**Comment by emmanuelkring** [comment]
I have made some performance comparisons now. I only tested PokerProbability. I used 1000000 iterations in the simulation and and averaged over 100 runs. With Array it took on average 2775 milliseconds to complete a simulation and with ArraySeq it took 3588 milliseconds. So there seems to be a performance penalty but ArraySeq is also more annoying to work with in this case since there it no toArraySeq method. Instead I had to use collection.mutable.ArraySeq.from().
